lib/bitcrowd_ecto/changeset.ex

```diff
@@ -112,4 +112,137 @@ defmodule BitcrowdEcto.Changeset do
       add_error(changeset, field, "is not a valid url", validation: :format)
     end
   end
+
+  @doc """
+  Validates a field timestamp to be in the past, if present
+  """
+  @doc since: "0.6.0"
+  @spec validate_past_datetime(Ecto.Changeset.t(), atom, DateTime.t()) :: Ecto.Changeset.t()
+  def validate_past_datetime(changeset, field, now \\ DateTime.utc_now()) do
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(now, datetime) == :lt do
+      add_error(changeset, field, "must be in the past", validation: :date_in_past)
+    else
+      changeset
+    end
+  end
+
+  @doc """
+  Validates a field timestamp to be in the future, if present
+  """
+  @doc since: "0.6.0"
+  @spec validate_future_datetime(Ecto.Changeset.t(), atom, DateTime.t()) :: Ecto.Changeset.t()
+  def validate_future_datetime(changeset, field, now \\ DateTime.utc_now()) do
+    datetime = get_change(changeset, field)
+
+    if datetime && DateTime.compare(now, datetime) != :lt do
+      add_error(changeset, field, "must be in the future", validation: :date_in_future)
+    else
+      changeset
+    end
+  end
+
+  @doc """
+  Validates a field timestamp to be after the given one
+  """
+  @doc since: "0.6.0"
+  @spec validate_datetime_after(Ecto.Changeset.t(), atom, DateTime.t(), fun) :: Ecto.Changeset.t()
+  def validate_datetime_after(
+        changeset,
+        field,
+        reference_datetime,
+        formatter \\ &DateTime.to_string/1
```

Comment:
i feel like this should be a keyword option, wdyt?

Reply:
Right! I also moved `valid_orders` and `compare_fun` to be part of the `opts`.